store/src/main/java/org/apache/rocketmq/store/CommitLog.java

```diff
@@ -1482,11 +1482,15 @@ public static class MessageExtEncoder {
         private final ByteBuf byteBuf;
         // The maximum length of the message body.
         private final int maxMessageBodySize;
-
+        // The maximum length of the full message.
+        private final int maxMessageSize;
         MessageExtEncoder(final int maxMessageBodySize) {
             ByteBufAllocator alloc = UnpooledByteBufAllocator.DEFAULT;
-            byteBuf = alloc.directBuffer(maxMessageBodySize);
+            //Reserve 64kb for encoding buffer outside body
+            int maxMessageSize = maxMessageBodySize + 64 * 1024;
```

Comment:
IMO, variable parameters, which is properties and topic name, is limited and properties before write into commitLog, if this two item length exceed limit, it won't reach here.

and I think it is a little weird to limit length base on an auto scalable ByteBuf.

Reply:
Yes, get to this step, properties(max 32kb) and topic name(max 127 bytes) are limited. But the full message length still may exceed 4M(maxMessageBodySize).
About whether it is neccessary to do this change, it's worth discussing. I think, in the previous version, maxMessageBodySize is also used to limit buffer length. So from my point of view, I think it is necessary to add.